CI doesn't seem to agree:

```
2020-09-25T09:33:24.6407435Z [ERROR] Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.172 s <<< FAILURE! - in io.quarkus.devtools.commands.ListExtensionsTest
2020-09-25T09:33:24.6417389Z [ERROR] io.quarkus.devtools.commands.ListExtensionsTest.testListExtensionsWithoutAPomFile  Time elapsed: 0.015 s  <<< FAILURE!
2020-09-25T09:33:24.6421899Z java.lang.AssertionError: 
2020-09-25T09:33:24.6423001Z 
2020-09-25T09:33:24.6426061Z Expecting code to raise a throwable.
2020-09-25T09:33:24.6435250Z 	at io.quarkus.devtools.commands.ListExtensionsTest.testListExtensionsWithoutAPomFile(ListExtensionsTest.java:178)
2020-09-25T09:33:24.6455999Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-09-25T09:33:24.6457533Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-09-25T09:33:24.6459365Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-09-25T09:33:24.6460786Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-09-25T09:33:24.6463155Z 	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
2020-09-25T09:33:24.6465004Z 	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
2020-09-25T09:33:24.6467141Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
2020-09-25T09:33:24.6469291Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
2020-09-25T09:33:24.6471838Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
2020-09-25T09:33:24.6474628Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
2020-09-25T09:33:24.6477616Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
2020-09-25T09:33:24.6480216Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
2020-09-25T09:33:24.6483104Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
2020-09-25T09:33:24.6486214Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
2020-09-25T09:33:24.6489327Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
2020-09-25T09:33:24.6492366Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
2020-09-25T09:33:24.6494807Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
2020-09-25T09:33:24.6497103Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
2020-09-25T09:33:24.6505376Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
2020-09-25T09:33:24.6516288Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-25T09:33:24.6527585Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
2020-09-25T09:33:24.6541196Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
2020-09-25T09:33:24.6549074Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
2020-09-25T09:33:24.6564292Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-09-25T09:33:24.6585944Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-25T09:33:24.6596034Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
2020-09-25T09:33:24.6603634Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2020-09-25T09:33:24.6613151Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
2020-09-25T09:33:24.6622043Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-25T09:33:24.6638176Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
2020-09-25T09:33:24.6646028Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
2020-09-25T09:33:24.6651080Z 	at java.util.ArrayList.forEach(ArrayList.java:1259)
2020-09-25T09:33:24.6662152Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-09-25T09:33:24.6673684Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
2020-09-25T09:33:24.6684835Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-25T09:33:24.6695730Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
2020-09-25T09:33:24.6703107Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2020-09-25T09:33:24.6712451Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
2020-09-25T09:33:24.6722131Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-25T09:33:24.6731777Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
2020-09-25T09:33:24.6740685Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
2020-09-25T09:33:24.6745988Z 	at java.util.ArrayList.forEach(ArrayList.java:1259)
2020-09-25T09:33:24.6758341Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-09-25T09:33:24.6769406Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
2020-09-25T09:33:24.6778315Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-25T09:33:24.6788157Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
2020-09-25T09:33:24.6795367Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2020-09-25T09:33:24.6802642Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
2020-09-25T09:33:24.6809941Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-25T09:33:24.6818824Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
2020-09-25T09:33:24.6828103Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
2020-09-25T09:33:24.6841477Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
2020-09-25T09:33:24.6853521Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2020-09-25T09:33:24.6864191Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
2020-09-25T09:33:24.6874466Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
2020-09-25T09:33:24.6883362Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
2020-09-25T09:33:24.6891316Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
2020-09-25T09:33:24.6900029Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
2020-09-25T09:33:24.6909435Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
2020-09-25T09:33:24.6916320Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
2020-09-25T09:33:24.6922467Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
2020-09-25T09:33:24.6930530Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
2020-09-25T09:33:24.6939949Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
2020-09-25T09:33:24.6949839Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
2020-09-25T09:33:24.6962493Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
2020-09-25T09:33:24.6964393Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2020-09-25T09:33:24.6965928Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
2020-09-25T09:33:24.6967648Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
```

--- a/devtools/maven/src/main/java/io/quarkus/maven/QuarkusProjectMojoBase.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/QuarkusProjectMojoBase.java
@@ -2,6 +2,7 @@ package io.quarkus.maven;
 
 import java.io.IOException;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 
@@ -64,11 +65,20 @@ public abstract class QuarkusProjectMojoBase extends AbstractMojo {
         validateParameters();
 
         final MessageWriter log = new MojoMessageWriter(getLog());
-        final Path projectDirPath = project.getBasedir().toPath();
-        final BuildTool buildTool = QuarkusProject.resolveExistingProjectBuildTool(projectDirPath);
+        final Path projectDirPath = baseDir();
+        BuildTool buildTool = QuarkusProject.resolveExistingProjectBuildTool(projectDirPath);
+        if (buildTool == null) {
+            // it's not Gradle and the pom.xml not found, so we assume there is not project at all
+            buildTool = BuildTool.MAVEN;
+        }
         final QuarkusPlatformDescriptor platformDescriptor = resolvePlatformDescriptor(log);
 
-        doExecute(QuarkusProject.of(project.getBasedir().toPath(), platformDescriptor, buildTool), log);
+        doExecute(QuarkusProject.of(baseDir(), platformDescriptor, buildTool), log);
+    }
+
+    protected Path baseDir() {
+        return project == null || project.getBasedir() == null ? Paths.get("").normalize().toAbsolutePath()
+                : project.getBasedir().toPath();
     }
 
     private QuarkusPlatformDescriptor resolvePlatformDescriptor(final MessageWriter log) throws MojoExecutionException {

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenBuildFile.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenBuildFile.java
@@ -130,6 +130,9 @@ public class MavenBuildFile extends BuildFile {
     }
 
     private Model initModel() throws IOException {
+        if (!hasProjectFile(BuildTool.MAVEN.getDependenciesFile())) {
+            return null;
+        }
         byte[] content = readProjectFile(BuildTool.MAVEN.getDependenciesFile());
         return MojoUtils.readPom(new ByteArrayInputStream(content));
     }

--- a/integration-tests/devtools/src/test/java/io/quarkus/devtools/commands/ListExtensionsTest.java
+++ b/integration-tests/devtools/src/test/java/io/quarkus/devtools/commands/ListExtensionsTest.java
@@ -4,16 +4,14 @@ import static io.quarkus.maven.utilities.MojoUtils.readPom;
 import static io.quarkus.platform.tools.ToolsConstants.IO_QUARKUS;
 import static java.util.Arrays.asList;
 import static java.util.stream.Collectors.toMap;
-import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.IOException;
 import java.io.PrintStream;
-import java.io.UncheckedIOException;
 import java.nio.file.Files;
-import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.util.HashSet;
 import java.util.Map;
@@ -172,11 +170,19 @@ public class ListExtensionsTest extends PlatformAwareTestBase {
     }
 
     @Test
-    void testListExtensionsWithoutAPomFile() throws IOException {
+    void testListExtensionsWithoutAPomFile() throws Exception {
         final Path tempDirectory = Files.createTempDirectory("proj");
         final QuarkusProject project = QuarkusProject.maven(tempDirectory, getPlatformDescriptor());
-        assertThatExceptionOfType(UncheckedIOException.class).isThrownBy(() -> readByKey(project))
-                .withRootCauseInstanceOf(NoSuchFileException.class);
+        final Map<AppArtifactKey, AppArtifactCoords> installed = readByKey(project);
+        assertTrue(installed.isEmpty());
+        assertFalse(project.getPlatformDescriptor().getExtensions().isEmpty());
+
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try (final PrintStream printStream = new PrintStream(baos, false, "UTF-8")) {
+            new ListExtensions(project, MessageWriter.info(printStream)).execute();
+        }
+        final String output = baos.toString("UTF-8");
+        Assertions.assertTrue(output.contains("Current Quarkus extensions available:"));
     }
 
     private void addExtensions(QuarkusProject quarkusProject, String... extensions) throws Exception {

